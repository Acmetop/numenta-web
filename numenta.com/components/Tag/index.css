/* Numenta company website source code. Copyright Â© 2016 Numenta.*/
/* Full details in LICENSE.txt, or contact us at <http://numenta.com>.*/
/* This program is free software: you can redistribute it and/or modify it under*/
/* the terms of the GNU Affero General Public License as published by the Free*/
/* Software Foundation, either version 3 of the License, or (at your option) any*/
/* later version. This program is distributed in the hope that it will be*/
/* useful, but WITHOUT ANY WARRANTY; without even the implied warranty of*/
/* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero*/
/* General Public License for more details. You should have received a copy of*/
/* the GNU Affero General Public License along with this program. If not, see*/
/* <https://www.gnu.org/licenses/agpl.html>.*/

.tag {
  composes: br2 from 'tachyons';
  composes: fw4 from 'tachyons';
  composes: mh1 from 'tachyons';
  composes: relative from 'tachyons';
  composes: ttu from 'tachyons';
  composes: white from 'tachyons';
  composes: bg-color-blue-lite from '../../static/assets/css/theme.css';
  font-size: 0.7rem;
  letter-spacing: 0.05rem;
  padding: 0.05rem 0.35rem 0;
  top: -0.05rem;
}
