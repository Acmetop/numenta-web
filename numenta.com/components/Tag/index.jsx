// Numenta company website source code. Copyright Â© 2016 Numenta.
// Full details in LICENSE.txt, or contact us at <http://numenta.com>.
// This program is free software: you can redistribute it and/or modify it under
// the terms of the GNU Affero General Public License as published by the Free
// Software Foundation, either version 3 of the License, or (at your option) any
// later version. This program is distributed in the hope that it will be
// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
// General Public License for more details. You should have received a copy of
// the GNU Affero General Public License along with this program. If not, see
// <https://www.gnu.org/licenses/agpl.html>.

import React from 'react'

import styles from './index.css'


const Tag = ({children}) => (
  <span className={styles.tag}>
    {children}
  </span>
)

Tag.propTypes = {
  children: React.PropTypes.any.isRequired,
}

export default Tag
