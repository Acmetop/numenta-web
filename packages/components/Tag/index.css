/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

.tag {
  composes: br2 from 'tachyons';
  composes: f6 from 'tachyons';
  composes: fw6 from 'tachyons';
  composes: mh1 from 'tachyons';
  composes: ttu from 'tachyons';
  composes: white from 'tachyons';
  background-color: #0a90cb;
  padding: 0.05rem 0.35rem 0;
}
