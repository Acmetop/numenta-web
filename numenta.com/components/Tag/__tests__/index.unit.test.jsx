import React from 'react'
import renderer from 'react-test-renderer'

import Tag from '../../Tag'


describe('<Tag /> React component unit test suite', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <Tag>
        New!
      </Tag>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
