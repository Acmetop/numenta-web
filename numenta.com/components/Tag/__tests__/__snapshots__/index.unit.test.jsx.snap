exports[`<Tag /> React component unit test suite Renders correctly 1`] = `
<span
  className="tag">
  New!
</span>
`;
